client/api/src/client.rs

```diff
@@ -108,6 +108,24 @@ pub trait BlockBackend<Block: BlockT> {
 	}
 }
 
+/// Interface to provide access to preimported blocks.
+/// A preimported block is one that has been announced, checked and verified
+/// but not yet imported into the chain's backend.
+pub trait PreImportedBlockProvider<Block: BlockT> {
```

Comment:
Why do we need this? 

You are again dragging sync specific information about the full nod, while this is all network specific.

Why don't you wire this just together with the `BlockRequestHandler`?

Reply:
Think about the flow of data... we're verifying the block through `ChainSync` which doesn't have a direct link to `BlockRequestHandler` to communicate preimproted blocks. However, `ChainSync` does have an instance to the client to inform it about verified blocks. The `BlockRequestHandler` also asks the client for the block data so the client is the central to both components.

Reply:
Btw, i am not saying this is the best possible way... it's just that this is what can be achieved with the current architecture.

Reply:
> we're verifying the block through `ChainSync` which doesn't have a direct link to `BlockRequestHandler` to communicate preimproted blocks

Just because there isn't a direct link yet, doesn't mean that it is the best solution to wire everything through the client. When designing systems you always try to have as much data locality as possible, which is clearly not the solution to move stuff through the client. 

If you check the code on how the network is setup, you see that the `BlockRequestHandler` is initialized before initializing the `NetworkWorker` which initializes the `Protocol` and this initializes the `ChainSync`. If you now would create some struct `VerifiedBlocks` or whatever that is passed to the `BlockRequestHandler` and to `ChainSync`, you would keep all this logic completely in the network layer.

Reply:
I implemented your suggestion